main/php_network.h

```diff
@@ -280,7 +282,7 @@ PHPAPI int php_network_connect_socket(php_socket_t sockfd,
 	php_network_connect_socket((sock), (addr), (addrlen), 0, (timeout), NULL, NULL)
 
 PHPAPI php_socket_t php_network_bind_socket_to_local_addr(const char *host, unsigned port,
-		int socktype, long sockopts, zend_string **error_string, int *error_code
+		int socktype, long sockopts, long linger, zend_string **error_string, int *error_code
```

Comment:
I think it might be better to introduce something more generic that can be used for not just linger so we don't have to change this function if there are other values needed. It could be either completely opaque or just some pointer to union that can hold multiple types.